src/RuleSet/RuleSet.php

```diff
@@ -63,9 +64,40 @@ public function __construct(array $set = [])
             }
         }
 
+        $set = self::normalizeConfig($set);
+
         $this->rules = $this->resolveSet($set);
     }
 
+    /**
+     * @param array<string, array<string, mixed>|bool> $rulesConfig
+     *
+     * @return array<string, array<string, mixed>|bool>
+     */
+    public static function normalizeConfig(array $rulesConfig): array
+    {
+        $normalizedConfig = [];
+
+        foreach ($rulesConfig as $name => $config) {
+            $normalizedConfig[self::normalizeRuleName($name)] = $config;
+        }
+
+        return $normalizedConfig;
+    }
+
+    /**
+     * We need to convert FQCN to rule name in order to resolve final configuration, and allow overriding
+     * rules that don't use FQCN as their name. This is especially helpful for configs that include rulesets
+     * and then fine-tune some rules or disable them. Without this normalisation it would be required to always
+     * use only exact names in the config (FQCN only where FQCN is used as name).
+     */
+    public static function normalizeRuleName(string $name): string
+    {
+        return is_a($name, FixerInterface::class, true)
+            ? (new $name())->getName()
```

Comment:
if rule was provided in config as FQCN, did we run FixerNameValidator on it's name?
also, is it in tests somewhere ?